server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -673,6 +675,36 @@ public void testDeleteStaleClusterUUIDs() throws IOException {
         }
     }
 
+    public void testFileNames() {
+        final Index index = new Index("test-index", "index-uuid");
+        final Settings idxSettings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(IndexMetadata.SETTING_INDEX_UUID, index.getUUID())
+            .build();
+        final IndexMetadata indexMetadata = new IndexMetadata.Builder(index.getName()).settings(idxSettings)
+            .numberOfShards(1)
+            .numberOfReplicas(0)
+            .build();
+
+        String indexMetadataFileName = RemoteClusterStateService.indexMetadataFileName(indexMetadata);
+        assertThat(indexMetadataFileName.split(DELIMITER).length, is(4));
+        assertTrue(indexMetadataFileName.contains(INDEX_METADATA_FILE_PREFIX));
+        assertTrue(indexMetadataFileName.contains(RemoteStoreUtils.invertLong(indexMetadata.getVersion())));
```

Comment:
instead of contains could be split the filename using DELIMITER and check on the exact position where these values are supposed to be in filename?

similarly for ln:698 to 700

Reply:
Updated.